pyhicu.py

```diff
@@ -1,113 +1,162 @@
+        return previous_right + [Move.UP] + previous + [Move.RIGHT] + previous + [Move.DOWN] + previous_left
+
 
+def moves2coordinates(moves):
+    """
+    Converts a sequence of 'Move' enum members into a corresponding sequence of 2D coordinates.
 
-def moves2cordinates(moves):
+    Returns:
+    A list of coordinate tuples (x, y) representing the coordinates after each move in the sequence.
+    """
     coords = [(0, 0)]
 
     for move in moves:
         prev = coords[-1]
-
-        if move == "u":
-            current = (prev[0], prev[1] + 1)
-
-        elif move == "d":
-            current = (prev[0], prev[1] - 1)
-        
-        elif move == "r":
-            current = (prev[0] + 1, prev[1])
-
-        elif move == "l":
-            current = (prev[0] - 1, prev[1])
-        
-        else:
-            raise ValueError(f"Invalid move combination: move {move} does not exist.")
-        
+        delta = move.get_delta()
+        current = (prev[0] + delta[0], prev[1] + delta[1])
         coords.append(current)
-    
+
     return coords
 
 
-def hilbert_curve(iteration=0):
-    return moves2cordinates(hilbert_core(iteration=iteration))
+def hilbert_curve(iteration):
+    """Generates a 2D NumPy array of coordinates representing a Hilbert curve of a given iteration."""
+    return np.array(moves2coordinates(hilbert_core(iteration)))
 
 
-def hilbert_vis(it, fs, dc, lc, filename):
-    plt.figure(figsize=(fs, fs))
-    dot_size = 2 ** (1 / (it + 1))
-    coords = np.array(hilbert_curve(it))
-    plt.plot(coords[:, 0], coords[:, 1], c=lc, linewidth=dot_size / 1.5)
-    plt.scatter(coords[:, 0], coords[:, 1], marker="o", color=dc, s=dot_size)
+def visualize_hilbert_curve(iteration, fig_size, dot_color, line_color, filename,
+                            dot_size_factor=1.5, line_width_factor=1.5, dpi=300):
+    """Generates and visualizes a Hilbert curve of a given iteration, saving the plot to a file."""
+    plt.figure(figsize=(fig_size, fig_size))
+    dot_size = 2 ** (1 / (iteration + 1)) * dot_size_factor
+    coords = hilbert_curve(iteration)
+    plt.plot(coords[:, 0], coords[:, 1], color=line_color, linewidth=dot_size / line_width_factor)
+    plt.scatter(coords[:, 0], coords[:, 1], marker="o", color=dot_color, s=dot_size)
     plt.yticks([])
     plt.xticks([])
-    plt.savefig(filename, bbox_inches="tight", pad_inches=0, dpi=300)
+    try:
+        plt.savefig(filename, bbox_inches="tight", pad_inches=0, dpi=dpi)
+    except Exception as e:
+        print(f"Error saving file: {e}")
+    plt.close()
 
 
-if __name__ == "__main__":
+def main():
     args = parse_args()
-    iteration = args.iteration
-    figside = args.fs
-    dotcol = args.dc
-    linecol = args.lc
-    outp = args.output
+    visualize_hilbert_curve(
+        args.iteration,
+        args.fig_size,
+        args.dot_color,
+        args.line_color,
+        args.output,
+        args.dot_size_factor,
+        args.line_width_factor,
+        args.dpi
+    )
```

Comment:
you use args "as is" without any modifications, so what not apply them like this